### ISSUE TITLE
[Feature Request]: Move stash pages

### ISSUE THREAD

**Issue by locksoft** [diasurgical/devilutionX#6997]
### Feature Type

Quality of Life

### Describe

On Switch the quick navigation buttons in the stash interface don’t work. Not a big issue, as it’s still possible navigating the pages with hardware buttons.

But a welcome function would be the possibility of swapping and moving the single pages. I always find myself organizing the contents as better as possible, and having to manually move 100 items every time from a page to a different one is a big pain (better have unique items or elixirs, first? Weapons or armors? And so on…).

**Comment by FitzRoyX** [comment]
I agree that this is a bit of a problem for stashes in heavy loot games. As the game progresses and the amount of loot in the stash swells, you realize that you didn't have the foresight to organize things optimally. This can result in a situation later on where you need to do a LOT of manual item moving.

PlugY is a popular stash mod for d2, and it apparently does have a command system for inserting pages and changing the page index. However, this method seems obtuse and unfriendly to me. I don't understand why 4 extra buttons in red wouldn't be better than having the user type "/insertpage" in the talk window. The buttons would amount to "move page left", "move page right", "delete blank page", and "insert page".

If you look at the devx stash design, there isn't really any space to add these buttons without removing an item row. However, if we were to replace the gold counter with a pile of gold on page 1, there would be. I guess you'll have to decide if this is more valuable than the gold counter. I think trying to reduce the size of the buttons or draw buttons on the frame to get around this would be pretty ugly.

![newstash](https://github.com/diasurgical/devilutionX/assets/57089001/e788101c-bd61-480e-8df5-61f1694bea4f)

**Comment by julealgon** [comment]
> However, if we were to replace the gold counter with a pile of gold on page 1, there would be. I guess you'll have to decide if this is more valuable than the gold counter.

To be fair, having an explicit "Gold" item in there is much more consistent with the rest of the game in D1's case. I think this is a good change.
